bigbang/archive.py

```diff
@@ -340,3 +357,179 @@ def clean_footer(foot):
+
+    Parameters
+    ----------
+    name : str
+    archive_dir : str, default CONFIG.mail_path
+    mbox : bool, default False
+        If true, expects and opens an mbox file at this path
+
+    Returns
+    -------
+    data : pandas.DataFrame
+
+    """
+
+    if mbox:
+        return open_list_archives(name, archive_dir=archive_dir, mbox=True)
+
+    # a first pass at detecting if the string is a URL...
+    if not (name.startswith("http://") or name.startswith("https://")):
+        path = os.path.join(archive_dir, name + ".csv")
+
+        if os.path.exists(path):
+            data = pd.read_csv(path)
+            return data
+        else:
+            logging.warning("No data available at %s", path)
+    else:
+        # TODO: Should "get_list_name" be more general than mailman?
+        path = os.path.join(archive_dir, mailman.get_list_name(name) + ".csv")
+
+        if os.path.exists(path):
+            data = pd.read_csv(path)
+            return data
+        else:
+            logging.warning(
+                "No data found for %s. Check directory name and whether archives have been collected.",
+                name,
+            )
+
+
+def messages_to_dataframe(messages):
+    """
+    Turn a list of parsed messages into a dataframe of message data,
+    indexed by message-id, with column-names from headers.
+    """
+    # extract data into a list of tuples -- records -- with
+    # the Message-ID separated out as an index
+    # valid_messages = [m for m in messages if m.get()
+
+    pm = [
+        (
+            m.get("Message-ID"),
+            str(m.get("From")).replace("\\", " "),
+            str(m.get("Subject")),
+            get_date(m),
+            str(m.get("In-Reply-To")),
+            str(m.get("References")),
+            get_text(m),
+        )
+        for m in messages
+        if m.get("From")
+    ]
+
+    mdf = pd.DataFrame.from_records(
+        list(pm),
+        index="Message-ID",
+        columns=[
+            "Message-ID",
+            "From",
+            "Subject",
+            "Date",
+            "In-Reply-To",
+            "References",
+            "Body",
+        ],
+    )
+    mdf.index.name = "Message-ID"
+
+    return mdf
+
+
+def open_list_archives(
+    url: str,
+    archive_dir: str = CONFIG.mail_path,
+    mbox: bool = False,
+) -> pd.DataFrame:
+    """
+    Return a list of all email messages contained in the specified directory.
+
+    Parameters
+    -----------
+    url: str
+        the name of a subdirectory of the directory specified
+        in argument *archive_dir*. This directory is expected to contain
+        files with extensions .txt, .mail, or .mbox. These files are all
+        expected to be in mbox format-- i.e. a series of blocks of text
+        starting with headers (colon-separated key-value pairs) followed by
+        an email body.
+        TODO: this argument should be re-named. sometimes it's being called with
```

Comment:
If we can't fix this akwardness right now, that's alright, I guess, but maybe we should open an issue for it. 

But at this point `url` seems to be either: 1) the name of a subdirectory; 2) a path to a .mbox file, or 3) a full url that will be converted into a listname which will be used as the name of a subdirectory.

Reply:
What's a good generic term for those things?

It can't be:
 - path
 - list
 - ... name ?

archive_name?
list_name?
list_id ?